components/supervisor/pkg/supervisor/supervisor.go

```diff
@@ -260,6 +275,9 @@ func Run(options ...RunOption) {
 	terminateChildProcesses()
 
 	wg.Wait()
+
+	log.Debugf("going to exit with: %d", exitCode)
```

Comment:
If we want to keep the debug log I'd much prefer it be structured, i.e. `log.WithField("exitCode", exitCode).Debug("supervisor exit")`